ggml/CMakeLists.txt

```diff
@@ -85,7 +85,7 @@ option(GGML_SANITIZE_ADDRESS   "ggml: enable address sanitizer"   OFF)
 option(GGML_SANITIZE_UNDEFINED "ggml: enable undefined sanitizer" OFF)
 
 # instruction set specific
-if (GGML_NATIVE OR NOT GGML_NATIVE_DEFAULT)
+if (NOT MSVC AND (GGML_NATIVE OR NOT GGML_NATIVE_DEFAULT))
```

Comment:
I'm not sure I understand why the `NOT MSVC` check is necessary here. Can you clarify further and maybe an example of when it makes a difference?

Also, are you utilizing the `CMAKE_CROSSCOMPILING` variable in your builds?

Reply:
I am using:
`cmake -S . -B build/win-x64 -A x64` (for building x64 Windows version with MSVC) without options in env variables (except the path for msvc, etc).

This will produce:
`CMAKE_CROSSCOMPILING: FALSE`
`GGML_NATIVE_DEFAULT: ON`
`GGML_NATIVE: ON`
`INS_ENV: OFF`

Before checking the usage of AVX and setting it to false only if it is not available on the machine (without setting it to true otherwise):
![image](https://github.com/user-attachments/assets/b10c6e1d-6661-496a-9cff-83225b136cc1)

The above configuration would produce an AVX-enabled library on AVX-enabled host.

After adding that check, the default will now be OFF. 
It's not a problem for me being only the default and still will be able to override it, but I don't think it is expected for others.

I was trying to allow all of this:

- building AVX-enabled library on AVX-enabled host

- building NotAVX-enabled library on AVX-enabled host

- building NotAVX-enabled library on NotAVX-enabled host

- Building AVX-enabled library by default on AVX-Enabled Host

- Building NotAVX-enabled library by default on NotAVX-Enabled Host

if removing that default check, the 4th will be true:

- Building NotAVX-enabled library by default on AVX-Enabled Host


Your call on this one, I can change the default however you say.